src/pint/models/model_builder.py

```diff
@@ -532,6 +565,71 @@ def choose_model(self, param_inpar):
                 selected_cates[cate] = cp
         return selected_components, conflict_components, param_not_in_pint
 
+    def choose_binary_model(self, param_inpar, force_binary_model=None, allow_T2=False):
+        """Choose the BINARY model based on the parfile.
+
+        Parameters
+        ----------
+        param_inpar: dict
+            Dictionary of the unique parameters in .par file with the key is the
+            parfile line. :func:`parse_parfile` returns this dictionary.
+
+        force_binary_model : str
+            When set to some binary model, this will override the binary model
+            set in the parfile.
+
+        allow_T2 : True, False, or str optional
+            Whether to convert a T2 binary model to an appropriate underlying
+            binary model. Default is False, and will throw an error upon
+            encountering the T2 binary model. If True, the binary model will be
+            converted to the most appropriate PINT-compatible binary model.
+
+        Returns
+        -------
+        str
+            Name of the binary component
+
+        Note
+        ----
+        If the binary model does not have a PINT model (e.g. the T2 model), an
+        error is thrown with the suggested model that could replace it. If an
+        appropriate model cannot be found, no suggestion is given
+        """
+        binary = param_inpar["BINARY"][0]
+
+        # Guess what the binary model should be, regardless of binary parameter
+        binary_model_guesses = guess_binary_model(param_inpar)
+
+        # Allow for T2 model, gracefully
+        if force_binary_model is not None:
+            # TODO: also need to convert here? Not very clean flow here
+            binary = force_binary_model
+        elif binary == "T2" and allow_T2:
+            binary = binary_model_guesses[0]
+            log.info(f"Found T2 binary model. Gracefully converting T2 to: {binary}.")
+
+            # Make sure that DDK parameters are properly converted
+            convert_binary_model(param_inpar, force_binary_model=binary)
```

Comment:
Can this operation throw an exception? If yes, will catching it here and printing a more informative error message be useful?

Reply:
The exception should then already be earlier. I added a try/except block there